File: Load.m

	%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
	%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
	%% This file rearranges input file for inventor name clustering
	%% in the next stage.
	%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
	%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


	clear;
	fid=fopen('./_attribute_matrix','r');
	temp=textscan(fid,'%d %d %d','delimiter',',','bufsize',2999999);
	fclose(fid);
	X=sparse(double(temp{1}),double(temp{2}),double(temp{3}));
	clear temp;
	fid=fopen('./_attribute_metric','r');
	temp=textscan(fid,'%s %f %s %s %s %s %s','delimiter','\t','bufsize',2999999);
	fclose(fid);
	patno=temp{1};
	inventor=temp{3};
	assignee=temp{4};
	class=temp{5};
	lastname=temp{6};
	inventor_id=temp{7};
	clear temp;

	XX=X>0;
	XXt=XX';

	fid=fopen(sprintf('./_attribute_dictionary'),'r');
	words=textscan(fid,'%*d %s %*d','delimiter','\t','bufsize',9999999);
	fclose(fid);
	words=words{1};

	fid=fopen(sprintf('./_disambiguator_input.csv'),'r');
	temp=textscan(fid,'%s %s %s %s %*s %*s %*s %*s %*s %*s %*s %*s','delimiter','\t','bufsize',9999999);
	fclose(fid);
	key=temp{1};
	name1=temp{2};
	name2=temp{3};
	name3=temp{4};
	namex=name1;
	for i=1:length(name1)
		namex{i}=[name1{i} ' ' name2{i} ' ' name3{i}];
	end
	namex=regexprep(namex,'[\ ]{1,}',' ');
	namex=strtrim(namex);
	clear temp i;

File: load.jl
function load_attribute_matrix(directory=".")
    cols = readdlm(joinpath(directory, "_attribute_matrix"), ',', Int)
    sparse(cols[:, 1], cols[:, 2], cols[:, 3])
end

function load_attribute_metric(directory=".")
    cols = readdlm(joinpath(directory, "_attribute_metric"), '\t', String, quotes=false)

    result = Dict{String, Vector{String}}()
    result["patno"] = cols[:, 1]
    result["inventor"] = cols[:, 3]
    result["assignee"] = cols[:, 4]
    result["class"] = cols[:, 5]
    result["lastname"] = cols[:, 6]
    result["inventor_id"] = cols[:, 7]

    result
end

function load_attribute_dictionary(directory=".")
    cols = readdlm(joinpath(directory, "_attribute_dictionary"), '\t', quotes=false)
    cols[:, 2]
end

function load_disambiguator_input(directory=".")
    key = String[]
    namex = String[]

    open(joinpath(directory, "_disambiguator_input.csv")) do f
        for line in eachline(f)
            fields = [strip(s) for s in split(line, "\t")]
            push!(key, fields[1])
            push!(namex, join(fields[2:4], " "))
        end
    end

    map!(t -> strip(replace(t, r"\s+", " ")), namex)
    ["key" => key, "namex" => namex]
end
